### PR TITLE
Fix alignment of parallel timetable blocks

### LINKED ISSUE #2999: Parallel contribution block is not shown in the correct place
All the sessions have a session code set.

https://indico.cern.ch/event/466934/timetable/#20170706.detailed

timetablebug

### PR DESCRIPTION
**Pull request by mic4ael**
fixes #2999

### CODE REVIEW

File: indico/htdocs/js/indico/modules/timetable/timetable/Layout.js [hunk 1]
```diff
@@ -402,7 +401,7 @@ type("CompactLayoutManager", ["IncrementalLayoutManager"],
                  algData.grid.push([startMin/60%24, algData.topPx]);
              }
 
-             if (!algData.active) {
+             if (algData.active == 0) {
```
Comment (ThiefMaster) [inline indico/htdocs/js/indico/modules/timetable/timetable/Layout.js]: why this change? if it's bool-ish we actually want `!`; and if you want to compare against the integer it should be `===`

Comment (mic4ael) [reply indico/htdocs/js/indico/modules/timetable/timetable/Layout.js]: it is not bool-ish :P the name is misleading, this variable is kind of a counter

**Comment by ThiefMaster** [comment]
unfortunately this is now causing some misalignment in other places:

alignment

**Comment by mic4ael** [comment]
fixed (I think..)

**Comment by ThiefMaster** [comment]
Unfortunately not :/ (CHEP 2016 timetable)

image

**Comment by mic4ael** [comment]
Looks a bit better now (at least the order is preserved)

screenshot from 2017-10-31 10-48-24

**Comment by ThiefMaster** [comment]
IIRC this was still problematic and more likely to cause problems (especially with the high amount of edge cases in the timetable), so it should probably not be merged for now.

**Comment by mic4ael** [comment]
As far as I remember nobody has reported to me any problems with the latest fix for this. I asked a few times if anyone could take a look. Either nobody has done that or there is no problem ;)

File: CHANGES.rst [hunk 2]
```diff
@@ -236,6 +236,7 @@ Bugfixes
 - Disable caching when retrieving the list of unscheduled contributions
 - Process placeholders in the subject when emailing registrants
 - Fix Shibboleth login with non-ascii names (:issue:`3143`)
+- Fix incorrect order of session blocks inside timetable (:issue:`2999`)
```
Comment (ThiefMaster) [inline CHANGES.rst]: move to 2.2 changes

File: CHANGES.rst [hunk 3]
```diff
@@ -29,6 +29,7 @@ Internal Changes
 - Use webpack to build static assets
 - Add React+Redux for new frontend modules
 - Enable modern ES201x features
+- Fix incorrect order of session blocks inside timetable (:issue:`2999`)
```
Comment (ThiefMaster) [inline CHANGES.rst]: it's a bugfix, not an internal change (add the "Bugfixes" section after "Improvements" like we did for other versions)

**Pull request merged**